src/pages/home/report/ReportActionCompose.js

```diff
@@ -310,18 +351,18 @@ class ReportActionCompose extends React.Component {
      * @param {String} newComment
      */
     updateComment(newComment) {
-        this.textInput.setNativeProps({text: newComment});
-        this.setState({
-            isCommentEmpty: newComment.length === 0,
-        });
-
-        // Indicate that draft has been created.
         if (this.comment.length === 0 && newComment.length !== 0) {
+            if (this.state.isCommentEmpty) {
+                this.setState({isCommentEmpty: false});
+            }
             Report.setReportWithDraft(this.props.reportID.toString(), true);
         }
 
         // The draft has been deleted.
         if (newComment.length === 0) {
+            if (!this.state.isCommentEmpty) {
+                this.setState({isCommentEmpty: true});
+            }
```

Comment:
I didn't understand this part. Also, I would like to ask that your original solution is very different from your proposal. Why so?

Reply:
Because I realize `updateCommnet` is handler for `onChangeText` and put `setNativePropse(text: )` in there isn't doing anything because when user type a character setNativeProps is being called even though the textInput is already changed by user type (you don't need `setNativeProps` there). 

As for `setState`, each time user type a character `setState` is being called then `ReportActionCompose` is being rendered. I am guarding `setState` of `isCommentEmpty` so it is not being called unnecessarily.

https://stackoverflow.com/questions/24718709/reactjs-does-render-get-called-any-time-setstate-is-called

Here I am attaching performance improvement, the first compose box is build using `developement` build but  it is faster than second compose box that is current production release.

https://user-images.githubusercontent.com/1313124/153715911-6d68ef6f-c4e2-48e7-abba-72c54cc96aee.mp4

Reply:
Ok. I think we can refactor this into a small function so that it does not look confusing.
```js
setCommentEmpty(IsEmpty){
	  if(this.state.isCommentEmpty === IsEmpty}{
	  	return;
	  }
	  this.setState({isCommentEmpty: IsEmpty});
}
```

Reply:
All right.